src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
 
   /**
-   * Generate documentation only for a specific submodule.
+   * A local directory containing jsii assembly files that will
+   * comprise the type-system.
    *
-   * @default - Documentation is generated for all submodules.
+   * @default - the root package directory will be used.
    */
-  readonly submoduleName?: string;
+  readonly assembliesDir?: string;
 }
 
 /**
  * Render documentation pages for a jsii library.
  */
 export class Documentation {
-  private readonly assembly: reflect.Assembly;
-  private readonly submodule?: reflect.Submodule;
-
-  private readonly includeReadme: boolean;
-  private readonly includeApiReference: boolean;
-
-  private readonly transpile: Transpile;
-
-  constructor(options: DocumentationOptions) {
-    this.assembly = options.assembly;
-
-    switch (options.language) {
-      case 'python':
-        this.transpile = new PythonTranspile();
-        break;
-      case 'ts':
-        this.transpile = new TypeScriptTranspile();
-        break;
-      default:
-        throw new Error(
-          `Generating documentation for ${options.language} is not supported`,
-        );
-    }
 
-    if (options.submoduleName) {
-      this.submodule = this.findSubmodule(options.submoduleName);
+  /**
+   * Create a `Documentation` object for a remote package available in the npm registry.
+   */
+  public static async forRemotePackage(name: string, version: string, options?: DocumentationOptions): Promise<Documentation> {
+    return withTempDir(async (workdir: string) => {
+      const env = {
+        ...process.env,
+        // npm fails with EROFS if $HOME is read-only, even if it won't write there
+        HOME: os.tmpdir(),
+      };
+
+      await exec('npm install npm@7', {
+        cwd: workdir,
+        env,
+      });
+      await exec(`${workdir}/node_modules/.bin/npm install --ignore-scripts --no-bin-links --no-save ${name}@${version}`, {
+        cwd: workdir,
+        env,
+      });
+
+      return Documentation.forLocalPackage(path.join(workdir, 'node_modules', name), { ...options, assembliesDir: workdir } );
+    });
+  }
+
+  /**
+   * Create a `Documentation` object for a package available in the local file system.
+   */
+  public static async forLocalPackage(root: string, options?: FromLocalPackageDocumentationOptions): Promise<Documentation> {
+    const manifestPath = path.join(root, 'package.json');
+    if (!(await fs.pathExists(manifestPath))) {
+      throw new Error(`Unable to locate ${manifestPath}`);
     }
 
-    this.includeApiReference = options.apiReference ?? true;
-    this.includeReadme = options.readme ?? true;
+    // normally the assemblies are located in subdirectories
+    // of the root package dir (i.e ./node_modules)
+    const assembliesDir = options?.assembliesDir ?? root;
+
+    const manifest = JSON.parse(await fs.readFile(manifestPath, 'utf-8'));
+    return Documentation.forAssembly(manifest.name, assembliesDir);
```

Comment:
This is nitpick/style/optional... I have grown to like unstructuring assignments when I only care about a few properties of a possibly large object... It makes it "obvious" what I'm interested in up-front, and also allows `node` to manage memory a little better (no need to retain the whole object in memory!)

```suggestion
    const { name } = JSON.parse(await fs.readFile(manifestPath, 'utf-8'));
    return Documentation.forAssembly(name, assembliesDir);
```

Reply:
👍